test/test_mite_websockets.py

```diff
@@ -29,28 +29,25 @@ async def dummy_journey(ctx):
         await ctx.websocket.connect("wss://foo.bar")
 
     with patch("websockets.connect", new=connect_mock):
-        await dummy_journey(context)
+        wb = await dummy_journey(context)
 
     assert getattr(context, "websocket", None) is None
 
 
 @pytest.mark.asyncio
-async def test_mite_websocket_connect_and_send():
```

Comment:
Should we add a test that validates the sending behavior?  Something like
```
def my_test():
    # set up mock / patch / etc.
    ws = await ctx.websocket.connect("ws://example.com")
    connect_mock.assert_called_once_with("ws://example.com")
    await ws.send("ohai")
    connect_mock.return_value.send.assert_called_once_with("ohai")
```

Reply:
I can try to do something for it but the send and recv functions are from the websockets library. 
Would it be meaningful if I call the send or request and then assert what I decided to return?

Reply:
If the connect function is a mock, then its return value won't be a real websocket, but a mock.  That's the brilliance of the python mock library, you replace one thing with a mock and it cascades through the whole logic, making it easy to write tests. 🙂